### ISSUE TITLE
"pre" hook

### ISSUE THREAD

**Issue by lonoak** [ltb-project/self-service-password#485]
Hi,

  this is more of an enhancement request... if it makes sense for you as well (I'm open for discussion).

There's a "post" hook to be executed after a successful password change, but I'd like to propose the possibility of a "pre" hook, in order to check the password (whatever that means, apart from the password policy) previous to attempting to change it. I'm thinking in things such as preventing from password re-use (if directory does not support it), or additional password strength checking...

In short, this is something like a "pre" hook to process more flexible password policies.

Do you see room for this request to be implemented? 

Thanks,

Jose.

**Comment by coudot** [comment]
Pre hook is a new feature of 1.4 release which should be ready soon.

See https://self-service-password.readthedocs.io/en/latest/config_preposthook.html

**Comment by lonoak** [comment]
Hi Clément, sorry to reopen this... do you have an estimation on when 1.4 will see the light? If not, what branch should I look to find this feature? thanks!

**Comment by faust64** [comment]
1.4 should be there, soon.
You can find it in the master branch.

**Comment by coudot** [comment]
@lonoak 1.4 is now released.

**Comment by lonoak** [comment]
Thanks @coudot ! ... I'll try updating Spanish translation, I see it's missing...
